src/action.ts

```diff
@@ -123,40 +124,86 @@ export async function performAction(
 	tracker: string
 ): Promise<ActionResult> {
 	const { action, linkDir } = getRuntimeConfig();
-	let linkedFilesRoot: string | undefined;
 
-	if (linkDir) {
-		const linkedFilesRootResult = await linkAllFilesInMetafile(
+	// SAVES FIRST, PREVENTS LINKING THOUGH IN THIS ORDER
+	if (action === Action.SAVE) {
+		saveTorrentFile(tracker, getTag(searchee.name), newMeta);
+		const styledName = chalk.green.bold(newMeta.name);
+		const styledTracker = chalk.bold(tracker);
+		logger.info(`Found ${styledName} on ${styledTracker} - saved`);
+		return SaveResult.SAVED;
+	}
+
+	let destinationDir: string | undefined,
+		ogDownloadDir,
+		linkedFilesRootResult;
+
+	// SET THE CLIENTS SAVE PATH - PREVENTS ERROR IF DATADIR/LINKDIR IS SET
+	// BUT NOT THE TORRENT DATA DIRECTORY AND/OR LINKING THROWS OR IS NOT SET
+	if (searchee.infoHash) {
+		ogDownloadDir = await getClient().getDownloadDir(searchee);
+		destinationDir = ogDownloadDir.isOk()
+			? ogDownloadDir.unwrapOrThrow()
+			: undefined;
+	}
+	if (
+		linkDir &&
+		typeof destinationDir === "string" &&
+		existsSync(destinationDir)
```

Comment:
so this check that the destination dir exists, is that the main addition that's preventing error here? what was the scenario that causes it? I'm wondering if we can handle the error inside `linkAllFilesInMetafile` because that's already calling `getDownloadDir`.

Reply:
Basically preventing the situation where linkDir is defined from, for instance, a media library from radarr - but your actual torrent data folder is not. Thus, it would ENOENT since the path it gets from the client is non-existent in the container.

This is mostly relevant for docker containers.